driver/src/main/scala/org/mongodb/scala/gridfs/GridFSDownloadObservable.scala

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.scala.gridfs
+
+import java.nio.{Buffer, ByteBuffer}
+
+import org.mongodb.scala.{Completed, Observable, Observer, Subscription}
+
+case class GridFSDownloadObservable(gridFSDownloadStream: GridFSDownloadStream) extends Observable[ByteBuffer] {
```

Comment:
Am I right that all this will go away in 4.x when Scala driver is built on top of the Reactive driver?

Reply:
If the answer is yes, then LGTM!

Reply:
Yes!